server/lib/util.py

```diff
@@ -500,3 +503,89 @@ def gzip_compress_response(raw_content, is_json):
   if is_json:
     response.headers['Content-Type'] = 'application/json'
   return response
+
+
+def fetch_highest_coverage(parent_entity: str,
+                           child_type: str,
+                           variables: List[str],
+                           all_facets: bool,
+                           facet_ids: List[str] = None):
+  """
+  Fetches the latest available data with the best coverage for the given a
+  parent entity, child type, variables, and facets
```

Comment:
To clarify, is the "best coverage" on a per stat var basis?

Reply:
Yes, it's per stat var. So a request for multiple stat vars could come back with 2021 for the best coverage for stat var A and 2022 for the best coverage for stat var B